src/__stories__/custom-input-story.tsx

```diff
@@ -0,0 +1,269 @@
+                    </Text3>
+                </Box>
+            ) : null}
+        </div>
+    );
+};
+
+export const Default: StoryComponent = () => {
+    const inputRef = React.useRef<HTMLInputElement>();
+    const [options, setOptions] = React.useState(OPTIONS);
+    const [filterValue, setFilterValue] = React.useState('');
+    const [selectedValue, setSelectedValue] = React.useState('');
+
+    const filterOptions = (valueInput: string, openMenu: () => void, isMenuOpen: boolean) => {
+        if (!isMenuOpen) {
+            openMenu();
+        }
+        setFilterValue(valueInput);
+    };
+
+    const checkIfRightOption = () => {
+        if (filterValue === selectedValue) return;
+
+        if (OPTIONS.filter(({value}) => value === filterValue).length === 0) {
+            setFilterValue('');
+            setSelectedValue('');
+        }
+    };
+
+    React.useEffect(() => {
+        setOptions(OPTIONS.filter(({value}) => value.includes(filterValue)));
+    }, [filterValue]);
+
+    return (
+        <StorySection title="Custom input">
+            <Stack space={32}>
+                <Text2 regular>
+                    This is an example of how to create a custom form input based on Mistica's pieces. This
+                    component is like a select with autocomplete feature.
+                </Text2>
+                <Form onSubmit={() => {}}>
+                    <Menu
+                        renderTarget={({ref, onPress, isMenuOpen}) => (
+                            <TextField
+                                fullWidth
+                                name="autocomplete"
+                                label="Autocomplete"
+                                value={filterValue}
+                                ref={combineRefs(ref, inputRef)}
+                                onFocus={onPress}
+                                onPress={() => {
+                                    if (!isMenuOpen) {
+                                        onPress();
+                                    }
+                                }}
+                                onChangeValue={(value) => filterOptions(value, onPress, isMenuOpen)}
+                                onBlur={checkIfRightOption}
+                                autoComplete="off"
+                            />
+                        )}
+                        renderMenu={({ref, className, close}) => (
+                            <Options
+                                elementRef={ref}
+                                options={options}
+                                className={className}
+                                selectedValue={selectedValue}
+                                setFilterValue={setFilterValue}
+                                setSelectedValue={setSelectedValue}
+                                closeMenu={close}
+                            />
+                        )}
+                    />
+                </Form>
```

Comment:
Better extract this into a standalone component, for example `AutocompleteSelectField`

And, to complete the example, expose `useFieldProps` and use that hook in this component. Add a submit button and check that the component is registered and working as expected.